homeassistant/components/peblar/config_flow.py

```diff
@@ -76,6 +76,53 @@ async def async_step_user(
             errors=errors,
         )
 
+    async def async_step_reconfigure(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle reconfiguration of a Peblar device."""
+        errors = {}
+
+        reconfigure_entry = self._get_reconfigure_entry()
+        host = reconfigure_entry.data[CONF_HOST]
+
+        if user_input is not None:
+            host = user_input[CONF_HOST]
+            peblar = Peblar(
+                host=host,
+                session=async_create_clientsession(
+                    self.hass, cookie_jar=CookieJar(unsafe=True)
+                ),
+            )
+            try:
+                await peblar.login(password=user_input[CONF_PASSWORD])
+            except PeblarAuthenticationError:
+                errors[CONF_PASSWORD] = "invalid_auth"
+            except PeblarConnectionError:
+                errors[CONF_HOST] = "cannot_connect"
+            except Exception:  # noqa: BLE001
+                LOGGER.exception("Unexpected exception")
+                errors["base"] = "unknown"
+            else:
+                return self.async_update_reload_and_abort(
+                    reconfigure_entry,
+                    data_updates=user_input,
+                )
+
+        return self.async_show_form(
```

Comment:
```suggestion
        reconfigure_entry = self._get_reconfigure_entry()

        if user_input is not None:
            host = user_input[CONF_HOST]
            peblar = Peblar(
                host=host,
                session=async_create_clientsession(
                    self.hass, cookie_jar=CookieJar(unsafe=True)
                ),
            )
            try:
                await peblar.login(password=user_input[CONF_PASSWORD])
            except PeblarAuthenticationError:
                errors[CONF_PASSWORD] = "invalid_auth"
            except PeblarConnectionError:
                errors[CONF_HOST] = "cannot_connect"
            except Exception:  # noqa: BLE001
                LOGGER.exception("Unexpected exception")
                errors["base"] = "unknown"
            else:
                return self.async_update_reload_and_abort(
                    reconfigure_entry,
                    data_updates=user_input,
                )

        host = reconfigure_entry.data[CONF_HOST]
        return self.async_show_form(
```
Let's move it down below as it made me go "huh, do we use the same host?"

Reply:
That is not the same code. I want the host to retain from user input when the user is in the process of changing and encountered an error. Your change resets all input when a users makes an error.

Reply:
Oh I was not aware that was the use case

Reply:
Moved it down for clarity 👍